projects/dev.py

```diff
@@ -35,6 +35,7 @@
     "openedx_learning.core.publishing.apps.PublishingConfig",
     # Learning Contrib Apps
     "openedx_learning.contrib.media_server.apps.MediaServerConfig",
+    "openedx_learning.contrib.tagging.apps.TaggingConfig",
```

Comment:
Hmm.. this tagging app is supposed to be a "plugin", which to me means:

1. it's installed alongside [edx-django-utils](https://github.com/openedx/edx-django-utils/) whose [plugin utils](https://github.com/openedx/edx-django-utils/blob/master/edx_django_utils/plugins/README.rst) which ensures that it gets automatically added to the `INSTALLED_APPS` list in the right context
1. it can be installed separately into apps where needed, and

edx-django-utils is specifically tailored to the `lms` and `cms` apps in edx-platform, and doesn't have an equivalent on this app, so..

Maybe we need to deal with the "pluggability" aspect when it comes time to extract this app out for reuse outside of openedx-learning?

Reply:
@pomegranited Good point! I didn't account for `edx-django-utils` for this. I have taken into account that it had to be ["installed alongside learning-core as a dependency in the edx-platform"](https://github.com/openedx/openedx-learning/blob/main/docs/decisions/0004-content-tagging.rst?plain=1#L14)

Another option would be to build this app to be installed alongside edx-django-utils so that we can use them with the current open-edx content and install it as a direct dependency on learning-core when we used the learning-core content. The problem is that I don't know how complicated this approach is in the long run. 

> Maybe we need to deal with the "pluggability" aspect when it comes time to extract this app out for reuse outside of openedx-learning?

I will leave this as an open question

Reply:
> Maybe we need to deal with the "pluggability" aspect when it comes time to extract this app out for reuse outside of openedx-learning?

Yeah, I think this is something that we can address later. I don't think it needs to be a plugin-utils styled plugin, but more of a library-style app that handles tagging related concerns and is used by our other apps. But I do think this means that the tagging functionality modeling in this repo should be split up so that there is:

1. An app or group of tagging-related apps that is entirely self contained, i.e. has no foreign keys to anything in the `openedx_learning` package. If we're pretty confident that this is going to be its own top level thing that will eventually be used in other parts of the platform, we should give it its own top level package, e.g. `openedx_tagging` (or some other unique name that we can eventually register on PyPI).
2. The `openedx_learning.contrib.tagging` app, which would be where we apply tagging to `openedx_learning` models like `PublishableEntity` and `PublishableEntityVersion`. This would probably stay in this repo over the long term.

Reply:
> Another option would be to build this app to be installed alongside edx-django-utils so that we can use them with the current open-edx content and install it as a direct dependency on learning-core when we used the learning-core content. The problem is that I don't know how complicated this approach is in the long run.

Didn't see this reply when I wrote my message. I like this approach.

Reply:
> Another option would be to build this app to be installed alongside edx-django-utils 

I like this approach too, but it has the limitation that edx-django-utils is specifically tailored to edx-platform -- i.e., you can currently [only register plugins for "lms" and "cms"](https://github.com/openedx/edx-django-utils/blob/master/edx_django_utils/plugins/docs/how_tos/how_to_create_a_plugin_app.rst#manual-setup) -- so we'd need to enhance it to also register plugins for learning core.

So I'd rather not convert this into a plugin right now, but plan for it by respecting the encapsulation @ormsbee laid out above:

> move this to a new `openedx_tagging` package in this repo, sibling of `openedx_learning`
> `openedx_tagging` has no foreign keys to anything in the `openedx_learning` package
> `openedx_learning.contrib.tagging` contains any models that need to link the two packages.

@ChrisChV I'm happy to do this move as part of the data architecture ADR which is coming this week.

Reply:
I like the approach :+1: 

> @ChrisChV I'm happy to do this move as part of the data architecture ADR which is coming this week, ref https://github.com/openedx/modular-learning/issues/60

Sure, thanks @pomegranited :+1: